crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -57,8 +56,21 @@ impl Query {
     /// Range of checkpoints that the RPC has data available for (for data
     /// that can be tied to a particular checkpoint).
     async fn available_range(&self, ctx: &Context<'_>) -> Result<AvailableRange> {
-        let (first, last) = ctx.data_unchecked::<PgManager>().available_range().await?;
-        Ok(AvailableRange { first, last })
+        let CheckpointViewedAt(checkpoint_viewed_at) = ctx.data()?;
+        let cloned = *checkpoint_viewed_at;
```

Comment:
It should be safe to implement `Copy` on `CheckpointViewedAt` so you can do this:

```suggestion
        let CheckpointViewedAt(checkpoint_viewed_at) = *ctx.data()?;
```

P.S. I feel like you're trolling me by calling this `cloned` 😉

Reply:
(+ tweaks to the other examples below after adding Copy)

Reply:
hmm I've already derived Copy for CheckpointViewedAt, but the compiler still complains

Reply:
I think because `ctx.data()?` returns a reference, so the compiler wants me to dereference it when I use it elsewhere


and this particular instance is further complicated by moving the `checkpoint_viewed_at: u64` into the function:
```
        let CheckpointViewedAt(checkpoint_viewed_at) = ctx.data()?;
        let result = ctx
            .data_unchecked::<Db>()
            .execute(move |conn| consistent_range(conn, Some(*checkpoint_viewed_at)))
            .await
            .extend()?;
```

Reply:
Or alternatively, `let checkpoint_viewed_at = ctx.data::<CheckpointViewedAt>()?.0;`

Reply:
ah wait, i think better yet, `let CheckpointViewedAt(checkpoint_viewed_at) = *ctx.data()?;`

Reply:
oh im blind. im blind. LMAO. taht's what you suggested. im blind.